source/common/http/utility.cc

```diff
@@ -411,9 +411,23 @@ Utility::parseHttp1Settings(const envoy::config::core::v3::Http1ProtocolOptions&
     ret.header_key_format_ = Http1Settings::HeaderKeyFormat::Default;
   }
 
+  return ret;
+}
+
+Http1Settings
+Utility::parseHttp1Settings(const envoy::config::core::v3::Http1ProtocolOptions& config,
+                            bool hcm_stream_error_set,
+                            const Protobuf::BoolValue& hcm_stream_error) {
+  Http1Settings ret = parseHttp1Settings(config);
```

Comment:
I think this function just return the override_stream_error if set, and if not returns the HCM value no?

Reply:
Hm, isn't it the case that when this function is called the HCM value hasn't been initialised to its default value yet (in case of no explicit configuration)? See https://github.com/envoyproxy/envoy/blob/c555587abea7addf3b8b3989231826d7e8fcf8ed/source/extensions/filters/network/http_connection_manager/config.cc#L233-L234

I guess we could move that initialisation so that it happens earlier, and then simplify this function as per your comment. WDYT?